File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
export function decodedValue(arr: string[]): number {
  let res = '';
  for (let i = 0; i < 2; i++) {
    res += colorCode(arr[i])
  }
  return Number(res);
}

const colorCode = (color: string): number => {
  return COLORS.indexOf(color)
}

export const COLORS =
  [
    'black',
    'brown',
    'red',
    'orange',
    'yellow',
    'green',
    'blue',
    'violet',
    'grey',
    'white',
  ]

File: solutions/zig/resistor-color-duo/1/resistor_color_duo.zig
pub const ColorBand = enum { black, brown, red, orange, yellow, green, blue, violet, grey, white };
pub fn colorCode(colors: [2]ColorBand) usize {
    const val1 = @as(u8, @intFromEnum(colors[0]));
    const val2 = @as(u8, @intFromEnum(colors[1]));
    return (val1 * 10) + val2;
}
